### ISSUE TITLE
Develop JSON-LD context specification for OCFL

### ISSUE THREAD

**Issue by ahankinson** [OCFL/spec#41]
As per https://github.com/OCFL/spec/issues/1#issuecomment-406052044, point 3.

**Comment by zimeon** [comment]
I think it is not possible sensibly model in JSON-LD the [fixity](https://ocfl.io/draft/spec/#fixity), [manifest](https://ocfl.io/draft/spec/#manifest), and [state](https://ocfl.io/draft/spec/#version) blocks as currently defined. I'll consider first the `fixity` block as that is the most complex.

Using the same example data as in https://ocfl.io/draft/spec/#fixity:

```
"fixity": {
    "md5": {
        "184f84e28cbe75e050e9c25ea7f2e939": [ "v1/metadata/foo.xml" ],
        "c289c8ccd4bab6e385f5afdd89b5bda2": [ "v1/image.tiff" ],
        "d41d8cd98f00b204e9800998ecf8427e": [ "v1/empty.txt" ]
    },
    "sha1": {
        "66709b068a2faead97113559db78ccd44712cbf2": [ "v1/metadata/foo.xml" ],
        "b9c7ccc6154974288132b63c15db8d2750716b49": [ "v1/image.tiff" ],
        "da39a3ee5e6b4b0d3255bfef95601890afd80709": [ "v1/empty.txt" ]
    }
}
```

We can't use Index Maps because then the index is thrown away (and we have real information in "md5" etc.), and Id Maps don't make sense because "md5" has local context (it means "md5 digests for things in this object") rather than global so it doesn't make sense to prefix it in some way (e.g. with `@base` or local defn.) to get and id.

I can see two obvious way to model this data (below). Neither of these end up using JSON-LD 1.1 features so work in 1.0 also. Paste into https://json-ld.org/playground-dev/ to see the RDF created.

## Model 1 - fixity resource for each algorithm+digest pair

object (1)--->(+) fixity (1)--->(+) [ (1)digestAlgorithm, (1)digest, (+)file ]

```
{
  "@context": {
    "id": "@id",
    "ocfl": "https://ocfl.io/terms/",
    "fixity": {
      "@id": "ocfl:fixity"
    },
    "digestAlgorithm": "ocfl:digestAlgorithm",
    "digest": "ocfl:digest",
    "file": "ocfl:file"
  },


  "id": "ark:123/456",
  "fixity": [
    {
      "digestAlgorithm": "md5",
      "digest": "184f84e28cbe75e050e9c25ea7f2e939",
      "file": [ "v1/metadata/foo.xml" ]
    },
    {
      "digestAlgorithm": "md5",
      "digest": "c289c8ccd4bab6e385f5afdd89b5bda2",
      "file": [ "v1/image.tiff" ]
    },
    {
      "digestAlgorithm": "md5",
      "digest": "d41d8cd98f00b204e9800998ecf8427e",
      "file": [ "v1/empty.txt" ]
    },
    {
      "digestAlgorithm": "sha1",
      "digest": "66709b068a2faead97113559db78ccd44712cbf2",
      "file": [ "v1/metadata/foo.xml" ]
    },
    {
      "digestAlgorithm": "sha1",
      "digest": "b9c7ccc6154974288132b63c15db8d2750716b49",
      "file": [ "v1/image.tiff" ]
    },
    {
      "digestAlgorithm": "sha1",
      "digest": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
      "file": [ "v1/empty.txt" ]
    }
  ]
}
```

## Model 2 - fixity resource for each algorithm, with a set of child digest resources

object (1)--->(+) fixity (1)--->(+) [ (1)digestAlgorithm, (+)digestsResource ]

digestsResource (1)--->(+) [ (1)digest, (+)file ]

```
{
  "@context": {
    "id": "@id",
    "ocfl": "https://ocfl.io/terms/",
    "fixity": {
      "@id": "ocfl:fixity"
    },
    "digestAlgorithm": "ocfl:digestAlgorithm",
    "digest": "ocfl:digest",
    "digests": "ocfl:digests",
    "file": "ocfl:file"
  },
  
  
  "id": "ark:123/456",
  "fixity": [
    {
      "digestAlgorithm": "md5",
      "digests": [
        {
          "digest": "184f84e28cbe75e050e9c25ea7f2e939",
          "file": [ "v1/metadata/foo.xml" ]
        },
        {
          "digest": "c289c8ccd4bab6e385f5afdd89b5bda2",
          "file": [ "v1/image.tiff" ]
        },
        {
          "digest": "d41d8cd98f00b204e9800998ecf8427e",
          "file": [ "v1/empty.txt" ]
        }
      ]
    },
    {
      "digestAlgorithm": "sha1",
      "digests": [
        {
          "digest": "66709b068a2faead97113559db78ccd44712cbf2",
          "file": [ "v1/metadata/foo.xml" ]
        },
        {
          "digest": "b9c7ccc6154974288132b63c15db8d2750716b49",
          "file": [ "v1/image.tiff" ]
        },
        {
          "digest": "da39a3ee5e6b4b0d3255bfef95601890afd80709",
          "file": [ "v1/empty.txt" ]
        }
      ]
    }
  ]
}
```

## Use in `state` and `manifest`

The second model crosses over nicely for use with `state` and `manifest`, albeit with a rather remotely defined overall `digestAlgorithm` at the object level.

```
{
  "@context": {
    "id": "@id",
    "ocfl": "https://ocfl.io/terms/",
    "versions": {
      "@id": "ocfl:versions",
      "@container": "@list",
      "@type": "ocfl:Version"
    },
    "version": "ocfl:version",
    "fixity": {
      "@id": "ocfl:fixity"
    },
    "digestAlgorithm": "ocfl:digestAlgorithm",
    "state": "ocfl:state",
    "digest": "ocfl:digest",
    "file": "ocfl:file"
  },
  
  
  "id": "ark:123/456",
  "versions": [
    {
      "version": "v1",
      "state": [
        {
          "digest": "7dcc35...c31",
          "file": [ "metadata/foo.xml" ]
        },
        {
          "digest": "cf83e1...a3e",
          "file": [ "empty.txt", "empty2.txt" ]
        },
        {
          "digest": "ffccf6...62e",
          "file": [ "image.tiff" ]
        }
      ]
    }
  ]
}
```
So, to sum up: I think we need either to change how we express our inventory in JSON, or reconsider whether we need it to be JSON-LD (cf. https://github.com/OCFL/spec/issues/1 )

**Comment by rosy1280** [comment]
Per discussion on 10/03/2018: we would like to ditch JSON-LD and create a new JSON schema.  @zimeon will take on this task.

**Comment by zimeon** [comment]
Replaced by #190 #191. Closing.
